File: interview-practice/arrays/rotateImage/rotate_img.js
function rotateImage(a) {
	var rotatedImage = [];
	console.log(a);

	var k = 0;
	for(var i = 0; i < a.length; i++){
		for(var j = a.length -1; j  >= 0; j--){
			console.log("rotating");
			console.log(rotatedImage[i][k]);
			console.log(a[j][i]);
			k++;
		}k = 0;
	}
}

rotateImage([[1,2,3],[4,5,6],[7,8,9]]);

File: interview-practice/arrays/rotateImage/rotate_img.c
// Definition for arrays:
// typedef struct arr_##name {
//   int size;
//   type *arr;
// } arr_##name;
//
// arr_##name alloc_arr_##name(int len) {
//   arr_##name a = {len, len > 0 ? malloc(sizeof(type) * len) : NULL};
//   return a;
// }
//
//
arr_arr_integer rotateImage(arr_arr_integer a) {
    int k = 0;
    arr_arr_integer final_img = alloc_arr_arr_integer(a.size);
    for(int i = 0; i < a.size; i++){
        final_img.arr[i] = alloc_arr_integer(a.size);
    }
   for(int i = 0; i < a.size; i++){
       for(int j = a.size - 1; j >= 0 ; j--){
           final_img.arr[i].arr[k] = a.arr[j].arr[i];
           k++;
       }k = 0;
   }
    return final_img;
}

